verification/policy_file_processor.go

```diff
@@ -0,0 +1,338 @@
+	policies := make(map[string]*TrustPolicy)
+	for _, policy := range pd.TrustPolicies {
+		newPolicy := policy
+		policies[policy.Name] = &newPolicy
+	}
+	return policies
+}
+
+// overwritePolicies replaces TrustPolicies of PolicyDocument with the given
+// policies.
+func (pd *PolicyDocument) overwritePolicies(newPolicies map[string]*TrustPolicy) {
+	policies := make([]TrustPolicy, 0, len(newPolicies))
+	for _, policy := range newPolicies {
+		policies = append(policies, *policy)
+	}
+	pd.TrustPolicies = policies
+}
+
+// NewTrustPolicy creates a new TrustPolicy with speicified fields.
+func NewTrustPolicy(name string, scopes []string, level string, override map[string]string, stores []string, identities []string) *TrustPolicy {
+	return &TrustPolicy{
+		Name:           name,
+		RegistryScopes: scopes,
+		SignatureVerification: SignatureVerification{
+			Level:    level,
+			Override: override,
+		},
+		TrustStores:       stores,
+		TrustedIdentities: identities,
+	}
+}
+
+// LoadTrustPolicies loads policies from the given file.
+func LoadTrustPolicies(policyPath string) ([]*TrustPolicy, error) {
+	policyBytes, err := os.ReadFile(policyPath)
+	if err != nil {
+		return nil, err
+	}
+	policyBytes = bytes.Trim(policyBytes, " ")
+
+	version, err := extractPolicyVersion(policyBytes)
+	if err != nil {
+		return nil, err
+	}
+
+	return ParseTrustPolicies(policyBytes, version)
+}
+
+// extractPolicyVersion extracts version by parsing the policy bytes.
+func extractPolicyVersion(policyBytes []byte) (string, error) {
+	var policy map[string]interface{}
+	if err := json.Unmarshal(policyBytes, &policy); err != nil {
+		return "", err
+	}
+	version, ok := policy["version"]
+	if !ok {
+		return "", errors.New("version is not specified")
+	}
+	return version.(string), nil
+}
+
+// ParseTrustPolicies creates policies from the given bytes of specified version.
+// Current implementation only supports policy in version "1.0". New versions
+// MUST be supported if added.
+func ParseTrustPolicies(policyBytes []byte, version string) ([]*TrustPolicy, error) {
+	if version == "1.0" {
+		policyDocument := &PolicyDocument{}
+		if err := json.Unmarshal(policyBytes, policyDocument); err != nil {
+			return nil, err
+		}
+		return policyDocument.ListPolicies(), nil
+	}
+	return nil, fmt.Errorf("unsupported version: %v", version)
+}
+
+// mergePolicy merges given policies into a new one. Values of the override
+// policy will override those of the base one.
+func mergePolicy(base *TrustPolicy, override *TrustPolicy) *TrustPolicy {
+	policy := base.deepCopy()
+
+	if len(override.RegistryScopes) > 0 {
+		policy.RegistryScopes = override.RegistryScopes
+	}
+	if override.SignatureVerification.Level != "" {
+		policy.SignatureVerification.Level = override.SignatureVerification.Level
+	}
+	if override.SignatureVerification.Override != nil {
+		policy.SignatureVerification.Override = override.SignatureVerification.Override
```

Comment:
`Override` is a map. Just want to make sure we will replace the whole map instead of merging the fields.

Reply:
Good point, it should be merged instead of replaced. I'll update it.